sherlock/annotators/annotator.py

```diff
@@ -0,0 +1,24 @@
+from registrable import Registrable
+from typing import List
+
+from sherlock import Document
+from sherlock import DocumentProcessor
+
+
+class Annotator(Registrable, DocumentProcessor):
+    """
+    Superclass for Annotators that execute a specific task on
+    a Document (e.g. Named-Entity-Recognition or Binary
+    Relation Classification)
+    """
+
+    @classmethod
+    def from_pretrained(cls, path: str, **kwargs) -> "Annotator":
+        raise NotImplementedError("Annotator must implement 'from_pretrained'.")
+
+    def annotate_documents(self, documents: List[Document]) -> List[Document]:
```

Comment:
does it make sense to have annotate_documents/annotate_document defined here? what is the purpose of DocumentProcessor interface then?

Reply:
if there is no good reason, I'd suggest to leave those 2 methods in DocumentProcessor, and rename them to process_document(s). (which actually fits the "DocumentProcessor" class name even better than annotate or predict. a document processor does not necessarily need to annotate, it may just do text cleaning or some other analysis.

Reply:
I overlooked that somehow. Makes a lot of sense!